src/bidiMapper/modules/cdp/CdpTargetManager.ts

```diff
@@ -154,9 +156,18 @@ export class CdpTargetManager {
     switch (targetInfo.type) {
       case 'page':
       case 'iframe': {
-        if (targetInfo.targetId === this.#selfTargetId) {
+        if (this.#selfTargetId === targetInfo.targetId) {
+          // break to detach from the self target.
           break;
         }
+        // Mapper only needs one session per target. If we receive additional
+        // auto-attached sessions, that is very likely coming from custom CDP
+        // sessions.
+        if (this.#targetIdsToBeIgnoredByAutoAttach.has(targetInfo.targetId)) {
```

Comment:
Why are we doing this only for these type of targets?

Reply:
good point, we probably can do it for all sessions